tests/unit/adapters/osf-adapter-test.js

```diff
@@ -1,12 +1,401 @@
+    this.stub(node, 'resolveRelationship', () => {
+        return {
+            addCanonicalRecord: addCanonicalStub
+        };
+    });
+
+    Ember.run(() => {
+        node.save().then(() => {
+            saveStubs.forEach(s => assert.ok(s.called));
+            assert.ok(addCanonicalStub.calledTwice);
+            // Can't use calledWith because sinon's deepEqual creates
+            // infinite recursive calls when comparing the Ember DS.Models
+            assert.equal(addCanonicalStub.args[0][0], contributors[0]);
+            assert.equal(addCanonicalStub.args[1][0], contributors[1]);
+        }, () => {
+            // Fail
+            assert.ok(false);
+        });
+    });
+});
+
+test('#_addRelated defers to _doRelatedRequest and adds records to the parent\'s canonical state', function(assert) {
+    let node = FactoryGuy.make('node');
+    let institution = FactoryGuy.make('institution');
+    node.get('affiliatedInstitutions').pushObject(institution);
+
+    var doRelatedStub = this.stub(OsfAdapter.prototype, '_doRelatedRequest', () => {
+        return new Ember.RSVP.Promise(resolve => resolve());
+    });
+    var relation = node.resolveRelationship('affiliatedInstitutions');
+    relation.hasLoaded = true;
+    var addCanonicalStub = this.stub(relation, 'addCanonicalRecord');
+
+    Ember.run(() => {
+        node.save().then(() => {
+            assert.ok(doRelatedStub.called);
+            assert.ok(addCanonicalStub.calledOnce);
+            assert.ok(addCanonicalStub.calledWith(institution));
+        }, () => {
+            // Fail
+            assert.ok(false);
+        });
+    });
+});
+
+test('#_updateRelated defers to _doRelatedRequest, pushes the update response into the store, and updates the parent\'s canonicalState', function(assert) {
+    this.inject.service('store');
+    let store = this.store;
+
+    let node = FactoryGuy.make('node', 'hasContributors');
+    var contribs = node.get('contributors');
+    var contrib = contribs.objectAt(1);
+
+    contrib.set('bibliographic', !contrib.get('bibliographic'));
+
+    var doRelatedStub = this.stub(OsfAdapter.prototype, '_doRelatedRequest', () => {
+        return new Ember.RSVP.Promise(resolve => resolve({
+            data: [
+                // A slight hack-- ingore the value returned from _doRelatedRequest
+                true
+            ]
+        }));
+    });
+    var addCanonicalStub = this.stub();
+    this.stub(node, 'resolveRelationship', () => {
+        return {
+            addCanonicalRecord: addCanonicalStub
+        };
+    });
+    var pushStub = this.stub(store, 'push', () => contrib);
+    var normalizeStub = this.stub(store, 'normalize');
+
+    Ember.run(() => {
+        node.save().then(() => {
+            assert.ok(doRelatedStub.calledOnce);
+            assert.ok(addCanonicalStub.calledOnce);
+            assert.ok(pushStub.calledOnce);
+            assert.ok(normalizeStub.calledOnce);
+        }, () => {
```

Comment:
The [RSVP source code comments](https://github.com/tildeio/rsvp.js/blob/master/lib/rsvp/promise.js#L107) use the pattern above, and I've experienced weirdness with using the `catch` callback (maybe a transpilation issue?). Additionally, the usage here is pretty constrained, i.e. I'm positive no other code is depending on the resolution of this promise.

So unless you're really sure and feel strongly about this, I'd prefer to leave it as is.